internal/cmd/pomerium/pomerium.go

```diff
@@ -137,14 +143,14 @@ func setupAuthenticate(opt *config.Options, controlPlane *controlplane.Server) e
 	return nil
 }
 
-func setupAuthorize(opt *config.Options, controlPlane *controlplane.Server, optionsUpdaters *[]config.OptionsUpdater) error {
+func setupAuthorize(opt *config.Options, controlPlane *controlplane.Server, optionsUpdaters *[]config.OptionsUpdater) (*authorize.Authorize, error) {
 	if !config.IsAuthorize(opt.Services) {
-		return nil
+		return nil, nil
```

Comment:
IMHO, we shouldn't do this. The common pattern in Go is when error is nil, it will guarantee that the pointer returned is not nil. From the caller site, it makes no sense if we still have to check for non-nil server when error return nil.

Can we return a new Server instance instead, and have a config to make it Run returns immediately. So you don't have to guard like in https://github.com/pomerium/pomerium/pull/904/files#diff-6671c9cee71180e2fc5b0e137e44b3e6R116

Reply:
we can't return a new server instance because that would break non all-in-one mode. I will move the check out of the function